### PR TITLE
Node logging in moveit_core

### LINKED ISSUE #2502: Use node logging in moveit_core

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

fixes #2502

### CODE REVIEW

**Comment by tylerjw** [comment]
This one is still not ready... seeing weird test failures locally, will resume working on this tomorrow.

File: moveit_ros/warehouse/src/save_as_text.cpp [hunk 1]
```diff
@@ -56,9 +56,8 @@ static const std::string ROBOT_DESCRIPTION = "robot_description";
 typedef std::pair<geometry_msgs::msg::Point, geometry_msgs::msg::Quaternion> LinkConstraintPair;
 typedef std::map<std::string, LinkConstraintPair> LinkConstraintMap;
 
-using moveit::getLogger;
-
-void collectLinkConstraints(const moveit_msgs::msg::Constraints& constraints, LinkConstraintMap& lcmap)
+void collectLinkConstraints(const moveit_msgs::msg::Constraints& constraints, LinkConstraintMap& lcmap,
+                            const rclcpp::Logger& logger)
```
Comment (sea-bass) [inline moveit_ros/warehouse/src/save_as_text.cpp:60]: Why was this function in particular modified to accept a logger vs. using `moveit::get_logger()`?

**Comment by tylerjw** [comment]
The snake case are functions from ROS where they use the style god intended. The pascal case is from moveit's clang-tidy rules because we match ROS 1 style.

File: moveit_core/collision_detection/src/collision_octomap_filter.cpp [hunk 2]
```diff
@@ -129,16 +135,16 @@ int collision_detection::refineContactNormals(const World::ObjectConstPtr& objec
...
             }
           }
-          // RCLCPP_INFO(LOGGER, "Contact point at [%.3f, %.3f, %.3f], cell size %.3f, occupied cells
+          // RCLCPP_INFO(getLogger(), "Contact point at [%.3f, %.3f, %.3f], cell size %.3f, occupied cells
           // %d",
           //                          contact_point.x(), contact_point.y(), contact_point.z(), cell_size, count);
 
           // octree->getOccupiedLeafsBBX(node_centers, bbx_min, bbx_max);
-          // RCLCPP_ERROR(LOGGER, "bad stuff in collision_octomap_filter.cpp; need to port octomap
+          // RCLCPP_ERROR(getLogger(), "bad stuff in collision_octomap_filter.cpp; need to port octomap
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_core/collision_detection/src/collision_octomap_filter.cpp:147]: Maybe, but I didn't write these lines. I'd like just to leave them as I'm hoping to keep this PR to just clean up the logging.

File: moveit_core/collision_distance_field/src/collision_env_distance_field.cpp [hunk 3]
```diff
@@ -582,14 +581,14 @@ bool CollisionEnvDistanceField::getIntraGroupCollisions(const collision_detectio
...
             //            name_2.c_str(),
             //                      dist ,(*collision_spheres_1)[k].radius_
             //                      ,(*collision_spheres_2)[l].radius_);
             //            Eigen::Vector3d sc1 = (*sphere_centers_1)[k];
             //            Eigen::Vector3d sc2 = (*sphere_centers_2)[l];
-            //            RCLCPP_DEBUG(LOGGER,"sphere center 1:[ %f, %f, %f ], sphere
+            //            RCLCPP_DEBUG(logger_,"sphere center 1:[ %f, %f, %f ], sphere
             //            center 2: [%f, %f,%f ], lbdc size =
             //            %i",sc1[0],sc1[1],sc1[2],
             //                      sc2[0],sc2[1],sc2[2],int(gsr->link_body_decompositions_.size()));
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_core/collision_distance_field/src/collision_env_distance_field.cpp:594]: Same response as above, would like to just leave things I didn't write.

File: moveit_core/collision_detection/src/collision_octomap_filter.cpp [hunk 4]
```diff
@@ -151,7 +157,7 @@ int collision_detection::refineContactNormals(const World::ObjectConstPtr& objec
             if (divergence > allowed_angle_divergence)
             {
               modified++;
-              // RCLCPP_INFO(LOGGER, "Normals differ by %.3f, changing: [%.3f, %.3f, %.3f] -> [%.3f,
+              // RCLCPP_INFO(getLogger(), "Normals differ by %.3f, changing: [%.3f, %.3f, %.3f] -> [%.3f,
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_core/collision_detection/src/collision_octomap_filter.cpp:160]: Same response as above, would like to just leave things I didn't write.

File: moveit_core/exceptions/src/exceptions.cpp [hunk 5]
```diff
@@ -37,19 +37,19 @@
...
 ConstructException::ConstructException(const std::string& what_arg) : std::runtime_error(what_arg)
 {
-  RCLCPP_ERROR(LOGGER, "Error during construction of object: %s\nException thrown.", what_arg.c_str());
+  RCLCPP_ERROR(getLogger("exception"), "Error during construction of object: %s\nException thrown.", what_arg.c_str());
 }
 
 Exception::Exception(const std::string& what_arg) : std::runtime_error(what_arg)
 {
-  RCLCPP_ERROR(LOGGER, "%s\nException thrown.", what_arg.c_str());
+  RCLCPP_ERROR(getLogger("exception"), "%s\nException thrown.", what_arg.c_str());
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_core/exceptions/src/exceptions.cpp:53]: I like your suggestion here, updating to "moveit_exception".

File: moveit_core/utils/src/logger.cpp [hunk 6]
```diff
@@ -45,29 +45,14 @@ namespace moveit
...
+// `setNodeLoggerName` function.
+rclcpp::Logger& getGlobalRootLogger()
 {
   static rclcpp::Logger logger = [&] {
     // A random number is appended to the name used for the node to make it unique.
     // This unique node and logger name is only used if a user does not set a logger
-    // through the `setLogger` method to their node's logger.
-    auto name = fmt::format("moveit_{}", rsl::rng()());
+    // through the `setNodeLoggerName` method to their node's logger.
+    auto name = fmt::format("____moveit_{}", rsl::rng()());
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_core/utils/src/logger.cpp:55]: Another mistake, I left this in from some testing I was doing.

File: moveit_ros/benchmarks/src/BenchmarkExecutor.cpp [hunk 7]
```diff
@@ -62,9 +62,16 @@
...
 #undef max
 
-using moveit::getLogger;
 using namespace moveit_ros_benchmarks;
 
+namespace
+{
+rclcpp::Logger getLogger()
+{
+  return moveit::getLogger("BenchmarkExecutor");
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_ros/benchmarks/src/BenchmarkExecutor.cpp:71]: fixing these to make them consistant

File: moveit_ros/moveit_servo/src/utils/command.cpp [hunk 8]
```diff
@@ -40,10 +40,12 @@
...
 #include <moveit/utils/logger.hpp>
 
-using moveit::getLogger;
-
 namespace
 {
+rclcpp::Logger getLogger()
+{
+  return moveit::getLogger("servo_command");
+}
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_ros/moveit_servo/src/utils/command.cpp:48]: made it just "servo", is that what you were hoping for?

File: moveit_ros/perception/point_containment_filter/src/shape_mask.cpp [hunk 9]
```diff
@@ -81,8 +89,7 @@ point_containment_filter::ShapeHandle point_containment_filter::ShapeMask::addSh
     std::pair<std::set<SeeShape, SortBodies>::iterator, bool> insert_op = bodies_.insert(ss);
     if (!insert_op.second)
     {
-      RCLCPP_ERROR(moveit::getLogger(),
-                   "Internal error in management of bodies in ShapeMask. This is a serious error.");
+      RCLCPP_ERROR(getLogger(), "Internal error in management of bodies in ShapeMask. This is a serious error.");
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_ros/perception/point_containment_filter/src/shape_mask.cpp:92]: lol, yea... no idea what is happening here

File: doc/MIGRATION_GUIDE.md [hunk 10]
```diff
@@ -13,29 +13,15 @@ However because of initaliziation order (child logger has to be created after rc
...
-You'll find this in the constructor of many of our classes.
-
-```C++
-, logger_(moveit::makeChildLogger("servo"))
-```
-
-Once you have a child logger you can use it in logging macros:
-```C++
-RCLCPP_INFO(logger_, "Very important info message");
+RCLCPP_INFO(moveit::getLogger("my_namespace"), "Very important info message");
```
Comment (v4hn) [inline doc/MIGRATION_GUIDE.md:24]: Honestly very stupid question:
```suggestion
moveit::logInfo("my_namespace", "Very important info message");
```
or
```suggestion
MOVEIT_INFO("my_namespace", "Very important info message");
```
do read loads better, right?

I would clearly support a readable wrapper over an awkward combination of MoveIt internal static magic (definitely not what god intended) TOGETHER with C macro magic (which was clearly invented by demons).

Comment (tylerjw) [reply doc/MIGRATION_GUIDE.md:24]: Here is the issues I see with those two approaches.

## functions

The macros capture the file, function, and line number from the place where the macro is called. If we wrapped the ros logging system in our own functions we would loose that.

## our own macros

This is a more ergonomic solution but would require me to write macros that wrap the ros macros. That would be a lot of code to write, test, and maintain. I think this approach is better given that tradeoff.

Comment (v4hn) [reply doc/MIGRATION_GUIDE.md:24]: Fair enough. We could probably limit a convenience macro wrapper to the plain variants for almost all uses. [That approach works well in OMPL](https://ompl.kavrakilab.org/group__logging.html#ga576d0bc79b521f19c5415f330e2b173d) and surely others. But then the question would be whether stream or printf-style and combinations might quickly blow up again if people disagree.

Still, an argument slot that only ever contains one generic term in correct usage should not exist either.

Between a rock and a hard place by design.

Comment (tylerjw) [reply doc/MIGRATION_GUIDE.md:24]: Yea, I think in this case the ROS 2 logging interface is just worse from a usability standpoint. I want to reduce the amount of abstractions I maintain so I don't want to maintain a set of wrapper logging macros. For that reason I think what is in this PR is the best compromise for now.

File: moveit_ros/move_group/src/default_capabilities/state_validation_service_capability.cpp [hunk 11]
```diff
@@ -43,7 +43,7 @@
 
 namespace move_group
 {
-MoveGroupStateValidationService::MoveGroupStateValidationService() : MoveGroupCapability("StateValidationService")
+MoveGroupStateValidationService::MoveGroupStateValidationService() : MoveGroupCapability("state_salidation_service")
```
Comment (sea-bass) [inline moveit_ros/move_group/src/default_capabilities/state_validation_service_capability.cpp:46]: typo

```suggestion
MoveGroupStateValidationService::MoveGroupStateValidationService() : MoveGroupCapability("state_validation_service")

**Review by sea-bass — APPROVED** [review]
just found 1 thing

**Pull request merged**